File: Wardrobe.java
// Wardrobe.java
import java.io.*;
import java.nio.file.*;
import java.time.*;
import java.util.*;
import java.util.stream.*;
import com.google.gson.*;

class ClothingItem {
    String id;
    String name;
    String category;
    String photo;
    String tags;
    String date;
    boolean favorite;
    String created_at;

    ClothingItem() {}
    ClothingItem(String name, String category, String photo, String tags, String date) {
        this.id = UUID.randomUUID().toString().substring(0,8);
        this.name = name;
        this.category = category;
        this.photo = photo;
        this.tags = tags;
        this.date = date.isEmpty() ? LocalDate.now().toString() : date;
        this.favorite = false;
        this.created_at = Instant.now().toString();
    }
}

class Wardrobe {
    private List<ClothingItem> items = new ArrayList<>();
    private final String dataFile = "wardrobe.json";
    private final Gson gson = new GsonBuilder().setPrettyPrinting().create();

    public Wardrobe() { load(); }

    private void load() {
        try {
            Path path = Paths.get(dataFile);
            if (Files.exists(path)) {
                String json = new String(Files.readAllBytes(path));
                ClothingItem[] arr = gson.fromJson(json, ClothingItem[].class);
                items = Arrays.asList(arr);
            }
        } catch (Exception e) {}
    }

    private void save() {
        try {
            Files.write(Paths.get(dataFile), gson.toJson(items).getBytes());
        } catch (Exception e) {}
    }

    public void add(String name, String category, String photo, String tags, String date) {
        ClothingItem item = new ClothingItem(name, category, photo, tags, date);
        items.add(item);
        save();
        System.out.printf("✅ Added: %s (ID: %s)%n", name, item.id);
    }

    public void list(String category) {
        List<ClothingItem> filtered = items;
        if (category != null && !category.isEmpty()) {
            filtered = items.stream()
                .filter(i -> i.category.equalsIgnoreCase(category))
                .collect(Collectors.toList());
        }
        if (filtered.isEmpty()) {
            System.out.println("No items.");
            return;
        }
        System.out.printf("\n📋 Wardrobe (%d items):\n", filtered.size());
        for (int i = 0; i < filtered.size(); i++) {
            ClothingItem item = filtered.get(i);
            String fav = item.favorite ? " ⭐" : "";
            String tags = item.tags != null && !item.tags.isEmpty() ? " | " + item.tags : "";
            System.out.printf("  %d. %s [%s]%s%s%n", i+1, item.name, item.category, fav, tags);
        }
    }

    public void search(String term) {
        String lower = term.toLowerCase();
        List<ClothingItem> results = items.stream()
            .filter(i -> i.name.toLowerCase().contains(lower) ||
                         i.category.toLowerCase().contains(lower) ||
                         (i.tags != null && i.tags.toLowerCase().contains(lower)))
            .collect(Collectors.toList());
        if (results.isEmpty()) {
            System.out.println("No matches.");
            return;
        }
        System.out.printf("\n🔍 Found %d item(s):\n", results.size());
        for (int i = 0; i < results.size(); i++) {
            ClothingItem item = results.get(i);
            String fav = item.favorite ? " ⭐" : "";
            String tags = item.tags != null && !item.tags.isEmpty() ? " | " + item.tags : "";
            System.out.printf("  %d. %s [%s]%s%s%n", i+1, item.name, item.category, fav, tags);
        }
    }

    public void show(String id) {
        for (ClothingItem item : items) {
            if (item.id.equals(id)) {
                System.out.printf("\n👔 %s%n", item.name);
                System.out.printf("  ID: %s%n", item.id);
                System.out.printf("  Category: %s%n", item.category.isEmpty() ? "Uncategorized" : item.category);
                System.out.printf("  Photo: %s%n", item.photo.isEmpty() ? "No photo" : item.photo);
                System.out.printf("  Tags: %s%n", item.tags.isEmpty() ? "None" : item.tags);
                System.out.printf("  Date: %s%n", item.date);
                System.out.printf("  Favorite: %s%n", item.favorite ? "Yes" : "No");
                System.out.printf("  Added: %s%n", item.created_at);
                return;
            }
        }
        System.out.printf("Item %s not found.%n", id);
    }

    public void favorite(String id) {
        for (ClothingItem item : items) {
            if (item.id.equals(id)) {
                item.favorite = !item.favorite;
                save();
                String status = item.favorite ? "⭐ added to" : "❌ removed from";
                System.out.printf("✅ %s favorites: %s%n", status, item.name);
                return;
            }
        }
        System.out.printf("Item %s not found.%n", id);
    }

    public void stats() {
        if (items.isEmpty()) {
            System.out.println("No items.");
            return;
        }
        Map<String, Integer> categories = new HashMap<>();
        for (ClothingItem item : items) {
            String cat = item.category.isEmpty() ? "Uncategorized" : item.category;
            categories.put(cat, categories.getOrDefault(cat, 0) + 1);
        }
        System.out.println("\n📊 Statistics:");
        categories.entrySet().stream()
            .sorted(Map.Entry.comparingByKey())
            .forEach(e -> System.out.printf("  %s: %d item(s)%n", e.getKey(), e.getValue()));
    }

    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Usage: Wardrobe <command> [options]");
            return;
        }
        Wardrobe w = new Wardrobe();
        String cmd = args[0];
        Map<String, String> params = new HashMap<>();
        for (int i=1; i<args.length; i++) {
            if (args[i].startsWith("--") && i+1 < args.length) {
                params.put(args[i].substring(2), args[++i]);
            }
        }
        switch (cmd) {
            case "add":
                if (args.length < 2) { System.out.println("add <name> [--category CAT] [--photo PATH] [--tags TAGS] [--date DATE]"); return; }
                w.add(args[1], params.getOrDefault("category", ""),
                      params.getOrDefault("photo", ""),
                      params.getOrDefault("tags", ""),
                      params.getOrDefault("date", ""));
                break;
            case "list":
                w.list(params.get("category"));
                break;
            case "search":
                if (args.length < 2) { System.out.println("search <term>"); return; }
                w.search(args[1]);
                break;
            case "show":
                if (args.length < 2) { System.out.println("show <id>"); return; }
                w.show(args[1]);
                break;
            case "favorite":
                if (args.length < 2) { System.out.println("favorite <id>"); return; }
                w.favorite(args[1]);
                break;
            case "stats":
                w.stats();
                break;
            default:
                System.out.println("Unknown command. Use add, list, search, show, favorite, stats.");
        }
    }
}

File: wardrobe.go
// wardrobe.go
package main

import (
	"encoding/json"
	"flag"
	"fmt"
	"os"
	"strings"
	"time"
	"github.com/google/uuid"
)

type ClothingItem struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	Category  string `json:"category"`
	Photo     string `json:"photo"`
	Tags      string `json:"tags"`
	Date      string `json:"date"`
	Favorite  bool   `json:"favorite"`
	CreatedAt string `json:"created_at"`
}

type Wardrobe struct {
	Items []ClothingItem `json:"items"`
}

var dataFile = "wardrobe.json"

func (w *Wardrobe) load() {
	data, err := os.ReadFile(dataFile)
	if err != nil {
		return
	}
	json.Unmarshal(data, w)
}

func (w *Wardrobe) save() {
	data, _ := json.MarshalIndent(w, "", "  ")
	os.WriteFile(dataFile, data, 0644)
}

func (w *Wardrobe) add(name, category, photo, tags, date string) {
	if date == "" {
		date = time.Now().Format("2006-01-02")
	}
	item := ClothingItem{
		ID:        uuid.New().String()[:8],
		Name:      name,
		Category:  category,
		Photo:     photo,
		Tags:      tags,
		Date:      date,
		Favorite:  false,
		CreatedAt: time.Now().Format(time.RFC3339),
	}
	w.Items = append(w.Items, item)
	w.save()
	fmt.Printf("✅ Added: %s (ID: %s)\n", name, item.ID)
}

func (w *Wardrobe) list(category string) {
	items := w.Items
	if category != "" {
		var filtered []ClothingItem
		for _, i := range items {
			if strings.EqualFold(i.Category, category) {
				filtered = append(filtered, i)
			}
		}
		items = filtered
	}
	if len(items) == 0 {
		fmt.Println("No items.")
		return
	}
	fmt.Printf("\n📋 Wardrobe (%d items):\n", len(items))
	for idx, item := range items {
		fav := ""
		if item.Favorite {
			fav = " ⭐"
		}
		tags := ""
		if item.Tags != "" {
			tags = " | " + item.Tags
		}
		fmt.Printf("  %d. %s [%s]%s%s\n", idx+1, item.Name, item.Category, fav, tags)
	}
}

func (w *Wardrobe) search(term string) {
	termLower := strings.ToLower(term)
	var results []ClothingItem
	for _, item := range w.Items {
		if strings.Contains(strings.ToLower(item.Name), termLower) ||
			strings.Contains(strings.ToLower(item.Category), termLower) ||
			strings.Contains(strings.ToLower(item.Tags), termLower) {
			results = append(results, item)
		}
	}
	if len(results) == 0 {
		fmt.Println("No matches.")
		return
	}
	fmt.Printf("\n🔍 Found %d item(s):\n", len(results))
	for idx, item := range results {
		fav := ""
		if item.Favorite {
			fav = " ⭐"
		}
		tags := ""
		if item.Tags != "" {
			tags = " | " + item.Tags
		}
		fmt.Printf("  %d. %s [%s]%s%s\n", idx+1, item.Name, item.Category, fav, tags)
	}
}

func (w *Wardrobe) show(id string) {
	for _, item := range w.Items {
		if item.ID == id {
			fmt.Printf("\n👔 %s\n", item.Name)
			fmt.Printf("  ID: %s\n", item.ID)
			fmt.Printf("  Category: %s\n", item.Category)
			fmt.Printf("  Photo: %s\n", item.Photo)
			fmt.Printf("  Tags: %s\n", item.Tags)
			fmt.Printf("  Date: %s\n", item.Date)
			fmt.Printf("  Favorite: %v\n", item.Favorite)
			fmt.Printf("  Added: %s\n", item.CreatedAt)
			return
		}
	}
	fmt.Printf("Item %s not found.\n", id)
}

func (w *Wardrobe) favorite(id string) {
	for i := range w.Items {
		if w.Items[i].ID == id {
			w.Items[i].Favorite = !w.Items[i].Favorite
			w.save()
			status := "⭐ added to"
			if !w.Items[i].Favorite {
				status = "❌ removed from"
			}
			fmt.Printf("✅ %s favorites: %s\n", status, w.Items[i].Name)
			return
		}
	}
	fmt.Printf("Item %s not found.\n", id)
}

func (w *Wardrobe) stats() {
	if len(w.Items) == 0 {
		fmt.Println("No items.")
		return
	}
	categories := make(map[string]int)
	for _, item := range w.Items {
		cat := item.Category
		if cat == "" {
			cat = "Uncategorized"
		}
		categories[cat]++
	}
	fmt.Println("\n📊 Statistics:")
	for cat, count := range categories {
		fmt.Printf("  %s: %d item(s)\n", cat, count)
	}
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: wardrobe <command> [options]")
		return
	}
	w := &Wardrobe{}
	w.load()
	cmd := os.Args[1]

	switch cmd {
	case "add":
		addCmd := flag.NewFlagSet("add", flag.ExitOnError)
		name := addCmd.String("name", "", "")
		category := addCmd.String("category", "", "")
		photo := addCmd.String("photo", "", "")
		tags := addCmd.String("tags", "", "")
		date := addCmd.String("date", "", "")
		addCmd.Parse(os.Args[2:])
		if *name == "" && len(addCmd.Args()) > 0 {
			*name = addCmd.Args()[0]
		}
		if *name == "" {
			fmt.Println("add requires a name")
			return
		}
		w.add(*name, *category, *photo, *tags, *date)

	case "list":
		listCmd := flag.NewFlagSet("list", flag.ExitOnError)
		category := listCmd.String("category", "", "")
		listCmd.Parse(os.Args[2:])
		w.list(*category)

	case "search":
		if len(os.Args) < 3 {
			fmt.Println("search <term>")
			return
		}
		w.search(os.Args[2])

	case "show":
		if len(os.Args) < 3 {
			fmt.Println("show <id>")
			return
		}
		w.show(os.Args[2])

	case "favorite":
		if len(os.Args) < 3 {
			fmt.Println("favorite <id>")
			return
		}
		w.favorite(os.Args[2])

	case "stats":
		w.stats()

	default:
		fmt.Println("Unknown command. Use add, list, search, show, favorite, stats.")
	}
}
